### ISSUE TITLE
Reinforcement Learning Agent

### ISSUE THREAD

**Issue by denizs** [denizs/torchUp#2]
The initial implementation of our agent should be as modular as possible in order to simplify the extension of this little program by additional agents.

Hence, every agent should be derived from a base class called `BaseAgent`, which implements some basic and shared methods, which can be easily extended for the various algorithms.

**The `__init__` method should at least set the following attributes:**
* `use_replay_memory` - Should the agent use a replay memory?
* `memory_cap` - How many entries should fit into our replay memory?
* `sample_batch_size` - Which size should our random batch samples have?
* `gamma` - Which discount factor for future rewards should be used?
* `eps_decay` - Should `eps` decay during training?
* `eps_decay_start` - Start value for eps
* `eps_decay_stop` - Final value for eps
* `eps_decay_rate` - Decay rate of epsilon.
* `model` - The model that the agent should use. This is expected to be of
type `torch.nn.Module`
* `use_cuda` - A flag indicating, whether the agent should offload complex
tensor calculations onto the GPU.
* `env` - The env to use, defaulting to OpenAI gym's 'CartPole-v0' env.
* `loss_function` - The loss function the agent should use.
* `optimizer` - The optmizer the agent should use

**Comment by denizs** [comment]
Implemented in #5
